multiqc/modules/hicexplorer/hicexplorer.py

```diff
@@ -0,0 +1,232 @@
+                        '<br><br><b>Same fragment</b><br>'
+                        'These are read mates, facing inward, separated by up to 800 bp that do not have a restriction enzyme in between.'
+                        ' These read pairs are not valid Hi-C pairs.'
+                        '<br><br><b>Self circle</b><br>'
+                        'Self circles are defined as pairs within 25kb with \'outward\' read orientation'
+                        '<br><br><b>Self ligation</b><br>'
+                        'These are read pairs with a restriction site in between that are within 800 bp.'
+        )
+
+        self.add_section(
+            name='Contact distance',
+            anchor='hicexplorer_contact_distance',
+            plot=self.hicexplorer_contact_distance()
+        )
+
+        self.add_section(
+            name='Read orientation',
+            anchor='hicexplorer_read_orientation',
+            plot=self.hicexplorer_read_orientation()
+        )
+
+    def parse_logs(self, f):
+        """Parse a given HiCExplorer log file from hicBuildMatrix."""
+        data = {}
+        for l in f.splitlines():
+            # catch empty lines
+            if len(l) == 0:
+                continue
+            s = l.split("\t")
+            data_ = []
+            # catch lines with descriptive content: "Of pairs used:"
+            for i in s[1:]:
+                if len(i) == 0:
+                    continue
+                try:
+                    i.replace('(', '')
+                    i.replace(')', '')
+                    i.replace(',', '')
+
+                    data_.append(float(i))
+                except:
+                    data_.append(i)
+            if len(data_) == 0:
+                continue
+            if s[0].startswith('short range'):
+                s[0] = 'short range'
+            elif s[0].startswith('same fragment'):
+                s[0] = 'same fragment'
+            data[s[0]] = tuple(data_)
+        return data
+
+    def hicexplorer_basic_statistics(self):
+        """Create the general statistics for HiCExplorer."""
+        data = {}
+
+        for file in self.mod_data:
+            total_pairs = self.mod_data[file]['Pairs considered'][0]
+            data_ = {
+                'Pairs considered': self.mod_data[file]['Pairs considered'][0],
+
+                'Pairs used': self.mod_data[file]['Pairs used'][0] / total_pairs,
+                'Unmapped': self.mod_data[file]['One mate unmapped'][0] / total_pairs,
+                'Min rest. site distance': self.mod_data[file]['Min rest. site distance'][0],
+                'Max rest. site distance': self.mod_data[file]['Max rest. site distance'][0],
+            }
+            data[self.mod_data[file]['File'][0]] = data_
+        headers = OrderedDict()
+
+        headers['Pairs considered'] = {
+            'title': 'Number of pairs',
+            'modify': lambda x: int(x)
```

Comment:
Do you need the `int()` here? Nice to add `'format': '{:,.0f}'` to cut off decimal places if you're using integers too.

Better still is to use the MultiQC config behaviour to handle read counts here though, so that users can customise how to display the count. See the docs here: http://multiqc.info/docs/#number-base-multiplier

Also, please use `'shared_key': 'read_count'` to make the scale bars consistent with counts from other modules (eg. FastQC etc).

Reply:
NB: Depending on timing, the number formatting stuff may be handled automatically with the shared key - see currently open PR https://github.com/ewels/MultiQC/pull/607

Reply:
Yeah, just need `'shared_key': 'read_count'` now (`modify` and the others will be set automatically then).